diff.go

```diff
@@ -84,27 +68,32 @@ type DiffInfo struct {
 
 	Component ComponentName
 
-	// TODO in case we don't use the DiffInfo structure for displaying a plan to
-	// the user, we need to move Action and Reason out of this.
-	Action string
-	Reason string
-
+	Key string
 	Old string
 	New string
 }
 
+type DiffInfos []DiffInfo
+
+func (dis DiffInfos) ComponentNames() ComponentNames {
+	componentNames := ComponentNames{}
+
+	for _, di := range dis {
+		componentNames = append(componentNames, di.Component)
```

Comment:
If `di.Component == ""`, it will add an empty name to the `componentNames` list. I think that should be avoided.